docs/reference/modules/discovery.asciidoc

```diff
@@ -1,30 +1,805 @@
+--------------------------------------------------
+
+Like all node settings, it is also possible to specify the initial set of
+master nodes on the command-line that is used to start Elasticsearch:
+
+[source,bash]
+--------------------------------------------------
+$ bin/elasticsearch -Ecluster.initial_master_nodes=master-a,master-b,master-c
+--------------------------------------------------
+
+It is technically sufficient to set this on a single master-eligible node in
+the cluster, and only to mention that single node in the setting, but this
+provides no fault tolerance before the cluster has fully formed. It
+is therefore better to bootstrap using at least three master-eligible nodes.
+In any case, when specifying the list of initial master nodes, **it is vitally
+important** to configure each node with exactly the same list of nodes, to
+prevent two independent clusters from forming. Typically you will set this on
+the nodes that are mentioned in the list of initial master nodes.
+
+NOTE: In alpha releases, all listed master-eligible nodes are required to be
+  discovered before bootstrapping can take place. This requirement will be
+  relaxed in production-ready releases.
+
+WARNING: You must put exactly the same set of initial master nodes in each
+  configuration file (or leave the configuration empty) in order to be sure
+  that only a single cluster forms during bootstrapping and therefore to
+  avoid the risk of data loss.
+
+[float]
+==== Choosing a cluster name
+
+The `cluster.name` allows you to create multiple clusters which are separated
+from each other. Nodes verify that they agree on their cluster name when they
+first connect to each other, and if two nodes have different cluster names then
+they will not communicate meaningfully and will not belong to the same cluster.
+The default value for the cluster name is `elasticsearch`, but it is
+recommended to change this to reflect the logical name of the cluster.
+
+[float]
+==== Auto-bootstrapping in development mode
+
+If the cluster is running with a completely default configuration then it will
+automatically bootstrap based on the nodes that could be discovered within a
+short time after startup. Since nodes may not always reliably discover each
+other quickly enough this automatic bootstrapping is not always reliable and
+cannot be used in production deployments.
+
+If any of the following settings are configured then auto-bootstrapping will
+not take place, and you must configure `cluster.initial_master_nodes` as
+described in the <<modules-discovery-bootstrap-cluster,section on cluster
+bootstrapping>>:
+
+* `discovery.zen.hosts_provider`
+* `discovery.zen.ping.unicast.hosts`
+* `cluster.initial_master_nodes`
+
+[float]
+[[modules-discovery-adding-removing-nodes]]
+=== Adding and removing nodes
+
+As nodes are added or removed Elasticsearch maintains an optimal level of fault
+tolerance by updating the cluster's _voting configuration_, which is the set of
```

Comment:
maybe add `automatically` to `updating`? it read to me like that's how I'm expected to do it.

Reply:
Sure, 7ca6cc8ce3d461475b34f8c95235011f346b8c5c.